core/consensus/consensus_event.hpp

```diff
@@ -53,32 +48,51 @@ class ConsensusEvent {
                 signature(sig)
         {}
     };
-    T tx;
+
+
     std::vector<eventSignature> eventSignatures;
 
 public:
     unsigned long long order = 0;
 
-    ConsensusEvent(T atx):
-        tx(atx)
-    {}
+    ConsensusEvent(
+        const std::string& senderPubkey,
+        const std::string& receiverPubkey,
+        const std::string& name,
+        const int& value
+    );
+    ConsensusEvent(
+        const std::string& senderPubkey,
+        const std::string& domain,
+        const std::string& name,
+        const unsigned long long& value,
+        const unsigned int& precision
+    );
+    ConsensusEvent(
+        const std::string& senderPubkey,
+        const std::string& ownerPublicKey,
+        const std::string& name
+    );
+
+
+    using Rule = json_parse::Rule;
+    using Type = json_parse::Type;
+    using Object = json_parse::Object;
+
+
 
     // WIP
-    ConsensusEvent(json_parse::Object obj) {}
+    ConsensusEvent(Object obj);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'ConsensusEvent' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478582987/issues/source?bid=3712491&fileBranchId=3923843#l85)